e2e/cypress/integration/files_and_attachments/upload_files_spec.js

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('Upload Files', () => {
+    let testTeam;
+
+    before(() => {
+        // # Create new team and new user and visit Town Square channel
+        cy.apiInitSetup({loginAfter: true}).then(({team}) => {
+            testTeam = team;
+            cy.visit(`/${testTeam.name}/channels/town-square`);
+        });
+    });
+
+    it('MM-T336 Image thumbnail - expanded RHS', () => {
+        const filename = 'huge-image.jpg';
+        const originalWidth = 1920;
+        const originalHeight = 1280;
+        const aspectRatio = originalWidth / originalHeight;
+        cy.log(aspectRatio);
+
+        // # Post an image in center channel
+        cy.get('#centerChannelFooter').find('#fileUploadInput').attachFile(filename);
+        cy.contains(filename).should('be.visible');
```

Comment:
I encountered failing test against remote server. The file details (filename, type, size) are shown immediately but file upload is not yet completed. My suggestion would be to check for `post-image` which indicates that the image has been uploaded and loaded into the UI and ready to be posted.
```suggestion
        cy.get('.post-image').should('be.visible');
```